### PR TITLE
Add tests for assignment and constructor

### PR DESCRIPTION
**Pull request by luckygalaxy666**
* Improve the unit tests to cover some parts.
* Correct some mistakes about creating a diag matirx in multithread as well.
* Correct mistake about construct a matrix with a specified value.

See #77.

### CODE REVIEW

**Review by Quentin9922 — APPROVED** [review]

**Comment by Kaiser-Yang** [comment]
Add periods here:
image

File: test/src/multi_thread_matrix_test.cpp [hunk 1]
```diff
@@ -383,5 +385,88 @@ TEST_F(TestMatrixMultiThread, powToOutput) {
     // make sure they are equal
     ASSERT_TRUE(equalSingleThread(singleOutput, multiOutput, 0, singleOutput.size()));
 }
+
+TEST_F(TestMatrixMultiThread, constructorFromInitializerList) {
+    Matrix<int> a({{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
```
Comment (Kaiser-Yang) [inline test/src/multi_thread_matrix_test.cpp:390]: remove this matrix a.

File: test/src/multi_thread_matrix_test.cpp [hunk 2]
```diff
@@ -383,5 +385,88 @@ TEST_F(TestMatrixMultiThread, powToOutput) {
     // make sure they are equal
     ASSERT_TRUE(equalSingleThread(singleOutput, multiOutput, 0, singleOutput.size()));
 }
+
+TEST_F(TestMatrixMultiThread, constructorFromInitializerList) {
```
Comment (Kaiser-Yang) [inline test/src/multi_thread_matrix_test.cpp:389]: rename the unit test with `constructorFromDiag`

File: test/src/multi_thread_matrix_test.cpp [hunk 3]
```diff
@@ -383,5 +385,88 @@ TEST_F(TestMatrixMultiThread, powToOutput) {
...
+    testing::Test::RecordProperty("BaseTime", executionTime / (threadNum() + 1));
+    startTime = high_resolution_clock::now();
+    Matrix<int> m1({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+    endTime       = high_resolution_clock::now();
+    executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+    // record time in gtest
+    testing::Test::RecordProperty("MultiTime", executionTime);
+
+    // TODO use operator== to check if they are equal
+    ASSERT_TRUE(equalSingleThread(a, n1, 0, n1.size()));
```
Comment (Kaiser-Yang) [inline test/src/multi_thread_matrix_test.cpp:420]: remove this.

File: test/src/multi_thread_matrix_test.cpp [hunk 4]
```diff
@@ -383,5 +385,88 @@ TEST_F(TestMatrixMultiThread, powToOutput) {
...
+    ASSERT_TRUE(equalSingleThread(m1, n1, 0, m1.size()));
+}
+
+TEST_F(TestMatrixMultiThread, assignmentFromInitializerList) {
+    // create a null matrix
+    a = std::initializer_list<std::initializer_list<double>>{};
+    // create a column null matirx
+    b                  = std::initializer_list<std::initializer_list<double>>{{}, {}};
```
Comment (Kaiser-Yang) [inline test/src/multi_thread_matrix_test.cpp:428]: add an empty line here to make b not align with the next line.

File: test/src/multi_thread_matrix_test.cpp [hunk 5]
```diff
@@ -383,5 +385,88 @@ TEST_F(TestMatrixMultiThread, powToOutput) {
...
+                     {0, 0, 0, 0, 0, 0, 0, 0, 8, 0},
+                     {0, 0, 0, 0, 0, 0, 0, 0, 0, 9}};
+    endTime       = high_resolution_clock::now();
+    executionTime = duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
+    // record time in gtest
+    testing::Test::RecordProperty("MultiTime", executionTime);
+
+    // TODO use operator== to check they are equal
+    ASSERT_TRUE(equalSingleThread(singleOutput, multiOutput, 0, singleOutput.size()));
+    ASSERT_EQ(a.shape().columns, 0);
```
Comment (Kaiser-Yang) [inline test/src/multi_thread_matrix_test.cpp:467]: `a.shape()` should be `Shape(0, 0)`.

File: test/src/multi_thread_matrix_test.cpp [hunk 6]
```diff
@@ -383,5 +385,88 @@ TEST_F(TestMatrixMultiThread, powToOutput) {
...
+    ASSERT_EQ(a.dataPtr(), nullptr);
+    ASSERT_EQ(b.dataPtr(), nullptr);
```
Comment (Kaiser-Yang) [inline test/src/multi_thread_matrix_test.cpp:469]: Check `b.shape()`.

**Review by Kaiser-Yang — CHANGES_REQUESTED** [review]

**Comment by Kaiser-Yang** [comment]
If you'd like them be seperated lines, bullet points are RECOMMENDED:
image

**Review by Kaiser-Yang — APPROVED** [review]

**Pull request merged**